lib/ret/http_utils.ex

```diff
@@ -1,31 +1,46 @@
 defmodule Ret.HttpUtils do
   use Retry
 
-  def retry_head_until_success(url, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:head, url, "", headers, cap_ms, expiry_ms)
+  def retry_head_until_success(url, options \\ []),
+    do: retry_until_success(:head, url, options)
 
-  def retry_get_until_success(url, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:get, url, "", headers, cap_ms, expiry_ms)
+  def retry_get_until_success(url, options \\ []),
+    do: retry_until_success(:get, url, options)
 
-  def retry_post_until_success(url, body, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:post, url, body, headers, cap_ms, expiry_ms)
+  def retry_post_until_success(url, body, options \\ []),
+    do: retry_until_success(:post, url, Keyword.merge(options, body: body))
 
-  def retry_put_until_success(url, body, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000),
-    do: retry_until_success(:put, url, body, headers, cap_ms, expiry_ms)
+  def retry_put_until_success(url, body, options \\ []),
+    do: retry_until_success(:put, url, Keyword.merge(options, body: body))
 
-  def retry_head_then_get_until_success(url, headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000) do
-    case url |> retry_head_until_success(headers, cap_ms, expiry_ms) do
+  def retry_head_then_get_until_success(url, options \\ []) do
+    case url |> retry_head_until_success(options) do
       :error ->
-        url |> retry_get_until_success(headers, cap_ms, expiry_ms)
+        url |> retry_get_until_success(options)
 
       res ->
         res
     end
   end
 
-  def retry_until_success(verb, url, body \\ "", headers \\ [], cap_ms \\ 5_000, expiry_ms \\ 10_000) do
+  defp retry_until_success(verb, url, options) do
+    default_options = [
+      body: "",
+      headers: [],
+      cap_ms: 5_000,
+      expiry_ms: 10_000,
+      append_browser_ua: false
```

Comment:
nit:  `ua` may be a difficult acronym to guess without looking at where it's used. Might be good to type it out. But not sure.